src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -538,6 +586,11 @@ protected static double dispatchQ(List<LfGenerator> generatorsWithControl, boole
             case K_EQUAL_PROPORTION -> allGeneratorsHavePlausibleReactiveLimits(generatorsWithControl)
                     ? splitDispatchQWithEqualProportionOfK(generatorsWithControl, qToDispatch)
                     : splitDispatchQ(generatorsWithControl, qToDispatch); // fallback to q dispatch
+            case REACTIVE_KEYS -> allGeneratorsHaveReactiveKeys(generatorsWithControl)
```

Comment:
I am not convinced by your fix. Indeed, during calculation, if reactive keys are present, we take them (`createReactiveKeys`). If not, we have two fall backs. `K_EQUAL_PROPORTION` is a bit special and must not be changed, but I think that the good solution is to adapt the `Q_EQUAL_PROPORTION` that seems to be too simple compared to what we do for calculation. To be confirmed by @geofjamg, but I think that you have to follow the same logic as the logic we have for creation.

Reply:
I agree with you, there is definetely something to discuss about my fix.
The thing is that reactive keys for calculations created by `createReactiveKeys` are used to split reactive power between **buses**. Whereas here `reactivePowerDispatchMode` controls how reactive power is split within a bus, between **generators**. I don't know if there are cases which we would like to split reactive power between buses in a certain way and in another way between generators...

Reply:
Yes of course, but as you said, we must be coherent...